### ISSUE TITLE
Program window not showing up with no error

### ISSUE THREAD

**Issue by JacekJagosz** [xournalpp/xournalpp#5198]
### Operating System

Linux

### (Linux only) Distribution

Solus 4.4

### (Linux only) Desktop Environment

KDE Plasma 5.27.8

### (Linux Only) Display Server

Wayland

### Installation Method

Solus repository

### Xournal++ Version

1.2.1

### libgtk Version

3.24.38

### Bug Description

When launching the app no window appears, but also no error log gets created in ~/.config/xournalpp.
When launching through terminal I get the following:
```
xopp-Message: 10:13:58.352: TEXTDOMAINDIR = (null), Platform-specific locale dir = /usr/share/xournalpp/../locale, chosen directory = /usr/share/xournalpp/../locale
ALSA lib pcm_dmix.c:999:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2666:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_route.c:877:(find_matching_chmap) Found no matching channel map
Cannot connect to server socket err = Nie ma takiego pliku ani katalogu
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
Cannot connect to server socket err = Nie ma takiego pliku ani katalogu
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
ALSA lib pcm_oss.c:397:(_snd_pcm_oss_open) Cannot open device /dev/dsp
ALSA lib pcm_oss.c:397:(_snd_pcm_oss_open) Cannot open device /dev/dsp
ALSA lib confmisc.c:160:(snd_config_get_card) Invalid field card
ALSA lib pcm_usb_stream.c:482:(_snd_pcm_usb_stream_open) Invalid card 'card'
ALSA lib confmisc.c:160:(snd_config_get_card) Invalid field card
ALSA lib pcm_usb_stream.c:482:(_snd_pcm_usb_stream_open) Invalid card 'card'
ALSA lib pcm_dmix.c:999:(snd_pcm_dmix_open) unable to open slave
xopp-Message: 10:13:58.433: Plugin "MigrateFontSizes" UI initialized
```
And the process keeps running after that, but with no window.

If I do `xournalpp --disable-audio` the output is much shorter:
```
xopp-Message: 10:14:19.553: TEXTDOMAINDIR = (null), Platform-specific locale dir = /usr/share/xournalpp/../locale, chosen directory = /usr/share/xournalpp/../locale
xopp-Message: 10:14:19.561: Plugin "MigrateFontSizes" UI initialized
```
The process again keeps running, but with no window

The same problem occurs on both my 2-in-1 laptop and my PC, both running Solus KDE with Xournalpp from the repos. I asked other Solus users and it works just fine for them.
The appimage of the same version works just fine.

How can I get any more in-depth logs to figure out what is wrong?

### Expected Behaviour

Window pops up when I launch the app

### Steps to Reproduce

1. Launch xournalpp

**Comment by rolandlo** [comment]
From the console output it looks like the `Control` instance was created successfully, but there must have been an issue afterwards. So the code has reached at least executed this line successfully:

https://github.com/xournalpp/xournalpp/blob/d6b7d5df081d36cac31f2d35e6764b860f0899af/src/core/control/XournalMain.cpp#L439-L440

Since there are usually no terminal messages from Xournal++'s side after that step it is hard to say, where the app failed.
If your version of Gtk is built with `G_ENABLE_DEBUG` (probably it isn't) you can run Xournal++ with `GTK_DEBUG=<option> xournalpp` or `GDK_DEBUG=<option> xournalpp`, where you find the options [here](https://docs.gtk.org/gtk3/running.html). 
Otherwise you probably need to [build from source](https://github.com/xournalpp/xournalpp/blob/master/readme/LinuxBuild.md) or attach a debugger to find the problem. Is any of these options fine for you? We can provide further assistance, of course.

**Comment by rolandlo** [comment]
You could also run Xournal++ via `GDK_BACKEND=x11 xournalpp`. Maybe it's a Wayland issue, so running it with the x11 backend might work fine.

**Comment by JacekJagosz** [comment]
It launches just fine with `GDK_BACKEND=x11 xournalpp`, so that at least narrows down the problem.

**Comment by bhennion** [comment]
Can you run other Gtk3 apps smoothly? For instance, does running `gtk3-demo` work as expected?

**Comment by JacekJagosz** [comment]
Yes it does. I know xournal works just fine on Gnome Wayland, so it must be KDE specific.

**Comment by tmoerschell** [comment]
No issue on my side with a very similar setup:

Operating System: openSUSE Tumbleweed 20231001
KDE Plasma Version: 5.27.8
KDE Frameworks Version: 5.110.0
Qt Version: 5.15.10
Kernel Version: 6.5.4-1-default (64-bit)
Graphics Platform: Wayland

Wouldn't be on OS side, right?

**Comment by bhennion** [comment]
Ah, I now remember having an issue like that once with Plasma Wayland (both with Libreoffice and Xournal++). The problem was that the window had size 0x0 so was not visible.
Could you try:

1. Launch the app - you see no window but the app should be visible in the taskbar
2. Right click on it in the taskbar -> More -> Resize
3. See the window.

**Comment by JacekJagosz** [comment]
I had that issue on KDE with other apps before, but unfortunately the app is not visible in the taskbar.

**Comment by JacekJagosz** [comment]
I finally got an error message:
```
(com.github.xournalpp.xournalpp:7325): Gdk-CRITICAL **: 12:24:38.530: gdk_wayland_window_set_dbus_properties_libgtk_only: assertion 'GDK_IS_WAYLAND_WINDOW (window)' failed
```

**Comment by tmoerschell** [comment]
I'd guess it's a KDE issue? 
Here's a similar chromium issue: https://bugs.chromium.org/p/chromium/issues/detail?id=1268758

You could try to play around with those settings in `~/.config/xournalpp/settings.xml`
```xml
  <property name="mainWndWidth" value="2752"/>
  <property name="mainWndHeight" value="1080"/>
  <property name="maximized" value="true"/>
```

**Comment by JacekJagosz** [comment]
Unfortunately changing the width, height, or setting maximized to true doesn't get rid of the problem. That was a good try though.
It really does seem like a weird problem with KDE

**Comment by sheepman4267** [comment]
I'm investigating this issue on Solus, per the issue where this was mentioned earlier today. Building from latest git source makes xournalpp work again on Plasma Wayland. Are there plans for a 1.2.3 release to address the commits made since 1.2.2?

**Comment by rolandlo** [comment]
> I'm investigating this issue on Solus, per the issue where this was mentioned earlier today. Building from latest git source makes xournalpp work again on Plasma Wayland. Are there plans for a 1.2.3 release to address the commits made since 1.2.2?

Although we haven't made plans yet in the dev team, I would support releasing 1.2.3 soon and I have enough spare time in the coming two weeks to do it.

**Comment by tmoerschell** [comment]
@sheepman4267, is tu issue fixed on the release-1.2 branch or only on master? I see no commit that could have fixed it over there.

**Comment by sheepman4267** [comment]
> @sheepman4267, is tu issue fixed on the release-1.2 branch or only on master? I see no commit that could have fixed it over there.

Well, now that you say it...

After I made my comment last night, I did a bit more testing, and you're right- it's only fixed on master. Apologies. Let me see if I can figure out what commit fixed it.

**Comment by JacekJagosz** [comment]
Did this commit never get into the 1.2.3? I have been running it in X11 mode all this time and still can't switch to Wayland when I run the package from distro's repository, that is 1.2.3.
Are there plans for a new release? Or what commit should be cherry-picked to fix this?

**Comment by fanzhuyifan** [comment]
Upstream kde bug report: https://bugs.kde.org/show_bug.cgi?id=491872

fwiw cannot reproduce with xournal++ 1.2.3, plasma git build, wayland, arch linux. xournal++ launches fine on my end, both through the desktop file, and through invoking xournalpp in konsole. I don't have the environment variable `GDK_BACKEND` set.

**Comment by ReillyBrogan** [comment]
I was able to bisect to find the commit that "fixed" this issue: 273a70767d7e5dcd7bad2a28285f0b4b4a6bff3d

It's a bit too complex to cherry-pick to v1.2.3 for me, so hopefully one of the xournalpp devs can help with that or suggest an alternate fix.

Also, I grabbed the stack trace from the crash. Looks like the CSS is infinitely recursing perhaps?
https://gist.github.com/ReillyBrogan/c9c0cb72b62e452e50456efcba6fde50
